tests/kafkatest/tests/consumer_test.py

```diff
@@ -292,28 +231,75 @@ def test_broker_failure(self, clean_shutdown, enable_autocommit):
+        """
+        consumer = self.setup_consumer(self.TOPIC)
+        producer = self.setup_producer(self.TOPIC)
 
-        consumer.start()
-        self._await_all_members(consumer)
+        partition = TopicPartition(self.TOPIC, 0)
 
         producer.start()
-        wait_until(lambda: producer.num_acked == total_records, timeout_sec=20,
-                   err_msg="Producer failed waiting for messages to be written")
 
-        wait_until(lambda: consumer.last_commit(partition) == total_records, timeout_sec=10,
-                   err_msg="Consumer failed to read all expected messages")
+        num_started = 0
+        for node in consumer.nodes:
+            consumer.start_node(node)
+            num_started += 1
+            self.await_members(consumer, num_started)
+            self.await_consumed_messages(consumer)
 
-        assert consumer.current_position(partition) == total_records
+        num_stopped = 0
+        for node in consumer.nodes:
+            consumer.stop_node(node)
+            num_stopped += 1
 
-    def test_valid_assignment(self):
-        consumer = self._setup_consumer(self.TOPIC)
-        consumer.start()
-        self._await_all_members(consumer)
-        assert self._valid_assignment(consumer.current_assignment())
+            if num_stopped < self.num_consumers:
+                self.await_members(consumer, self.num_consumers - num_stopped)
+                self.await_consumed_messages(consumer)
+
+        assert consumer.current_position(partition) == consumer.total_consumed(), \
+            "Total consumed records did not match consumed position"
+
+        assert consumer.last_commit(partition) == consumer.current_position(partition), \
+            "Last committed offset did not match last consumed position"
+
+
+class AssignmentValidationTest(VerifiableConsumerTest):
+    TOPIC = "test_topic"
+    NUM_PARTITIONS = 6
+
+    def __init__(self, test_context):
+        super(AssignmentValidationTest, self).__init__(test_context, num_consumers=3, num_producers=0,
+                                                num_zk=1, num_brokers=2, topics={
+            self.TOPIC : { 'partitions': self.NUM_PARTITIONS, 'replication-factor': 2 },
+        })
+
+    @matrix(assignment_strategy=["org.apache.kafka.clients.consumer.RangeAssignor",
+                                 "org.apache.kafka.clients.consumer.RoundRobinAssignor"])
+    def test_valid_assignment(self, assignment_strategy):
+        """
+        Verify assignment strategy correctness: each partition is assigned to exactly
+        one consumer instance.
+
+        Setup: single Kafka cluster with a set of consumers in the same group.
+
+        - Start the consumers one by one
+        - Validate assignment after every expected rebalance
+        """
+        consumer = self.setup_consumer(self.TOPIC, assignment_strategy=assignment_strategy)
+        num_started = 0
+        for node in consumer.nodes:
```

Comment:
Not critical, but `for num_started, node in enumerate(consumer.nodes, 1)` would probably be more idiomatic.